File: catkin_ws/src/my_robot_tutorials/src/hw_publisher.cpp
#include <ros/ros.h>
#include<my_robot_msgs/HardwareStatus.h>


int main(int argc, char** argv)
{
    ros::init(argc, argv, "hardware_status_publisher");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<my_robot_msgs::HardwareStatus>("/my_robot/hardware_status", 10);

    ros::Rate rate(5);
    my_robot_msgs::HardwareStatus hw_msg;

    hw_msg.are_motors_up=true;
    hw_msg.temperature=45.123;
    hw_msg.debug_msg="Motors are up";

    while(ros::ok())
    {
        pub.publish(hw_msg);
        rate.sleep();
    }
}

File: catkin_ws/src/my_robot_tutorials/scripts/hw_publisher.py
#! /usr/bin/env python

import rospy
from my_robot_msgs.msg import HardwareStatus

if __name__ == "__main__":
    rospy.init_node("hardware_status_publisher")

    pub = rospy.Publisher("/my_robot/hardware_status", HardwareStatus, queue_size=10)

    rate = rospy.Rate(5)

    while not rospy.is_shutdown():
        msg = HardwareStatus()
        msg.are_motors_up = True
        msg.temperature = 45
        msg.debug_msg = "Robot is running"

        pub.publish(msg)
        rate.sleep()